[11.0][MIG][WIP] base

rruebner opened the pull request:
Added first part / try of base module migration, it is not done yet. Missing parts are marked with "TODO" in openupgrade_analysis_work.txt.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

rruebner committed:
[11.0][MIG][WIP] base

rruebner committed:
[11.0][MIG][WIP] Add general module renaming

StefanRijnhart commented on odoo/addons/base/migrations/11.0.1.3/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,429 @@
+base         / ir.actions.server        / id_object (reference)         : DEL 
+base         / ir.actions.server        / id_value (char)               : DEL 
+base         / ir.actions.server        / link_new_record (boolean)     : DEL 
+base         / ir.actions.server        / menu_ir_values_id (many2one)  : DEL relation: ir.values
```
For actions with a menu_ir_values_id, set `binding_model_id` to action.model_id.id and `binding_type` to `action`. Delete the ir.values entry.

rruebner replied:
Done.

StefanRijnhart commented on odoo/addons/base/migrations/11.0.1.3/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,429 @@
+
+base         / ir.actions.server        / state (selection)             : selection_keys is now '['code', 'multi', 'object_create', 'object_write']' ('function')
+# DONE: pre-migration: copy column
+# TODO: post-migration: map values 'trigger' and 'client_action' to default value
```
Leave `trigger` value, so that it can be identified as invalid (there are no workflows anymore in Odoo 11). Leave client_action entries as well. They should be removed by the migration of ir.actions.server, and other entries should be recognizable as obsolete.

rruebner replied:
Done.

StefanRijnhart commented on odoo/addons/base/migrations/11.0.1.3/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,429 @@
+
+base         / ir.cron                  / interval_type (selection)     : selection_keys is now '['days', 'hours', 'minutes', 'months', 'weeks']' ('['days', 'hours', 'minutes', 'months', 'weeks', 'work_days']')
+# DONE: pre-migration: copy column
+# TODO: post-migration: map value 'work_days' to default value
```
How about mapping 'work_days' to 'days', as the default value is 'months'?

rruebner replied:
Updated todo text.

rruebner committed:
[11.0][MIG][WIP] Add first post migration to map first ir.actions.server values

StefanRijnhart commented on odoo/addons/base/migrations/11.0.1.3/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,428 @@
+
+base         / res.currency.rate        / name (datetime)               : type is now 'date' ('datetime')
+# DONE: pre-migration: copy column
+# DONE: pre-migration: change field type
```
I can't find where this happens. Am I missing something?

rruebner replied:
:) your are right. Its not done yet.

StefanRijnhart committed:
[RVT] Remove aggressive field copying. Data will be preserved anyway.
[IMP] Default values for new fields
[DEL] Irrelevant copy/paste section from field rename.
[ADD] Convert currency rate date per company's most popular timezone

StefanRijnhart commented:
Making some suggestions here: https://github.com/rruebner/OpenUpgrade/pull/1

rruebner committed:
Merge pull request #1 from StefanRijnhart/11.0-base_refactor

[RVT] Remove aggressive field copying. Data will be preserved anyway.

rruebner committed:
[ADD] Merge 'default' ir.values into ir.default.

rruebner commented:
@StefanRijnhart PR reviewed and merged, added 'default' ir.values merging into ir.default

StefanRijnhart commented:
Thank you!

hbrunn commented on odoo/addons/base/migrations/11.0.1.3/openupgrade_analysis_work.txt:
```diff
@@ -0,0 +1,399 @@
+# NOTHING TO DO, obsolete column
+
+base         / ir.actions.act_window    / binding_model_id (many2one)   : NEW relation: ir.model
+base         / ir.actions.act_window    / binding_type (selection)      : NEW required: required, selection_keys: ['action', 'report'], req_default: function
```
there's a lot to do for this: This fields (actually, this happens in the action's base model) replace the horrible ir.values model to link actions to models for the sidebar. You'll have to fill those values from the appropriate ir.values record

hbrunn commented on odoo/addons/base/migrations/11.0.1.3/post-migration.py:
```diff
@@ -0,0 +1,92 @@
+from psycopg2.extensions import AsIs
+
+
+def map_ir_actions_server_fields(cr):
```
as written above, you need to do this for all ir.actions.* models. Don't rely on the link from the action to the ir.value record, do it the other way around via https://github.com/OCA/OCB/blob/10.0/odoo/addons/base/ir/ir_values.py#L95, possibly filtering for key and key2

StefanRijnhart replied:
@hbrunn I addressed your comments in #1121 (except that ir.values' action_id is almost never filled and when it is, the key already provides the action reference as it does in all other instances)

StefanRijnhart commented:
Closing as review can continue on #1121
